scripts/transformer_qa_eval.py

```diff
@@ -0,0 +1,85 @@
+import json
+import logging
+import time
+from typing import Iterable, List
+
+from allennlp.common.checks import check_for_gpu
+from allennlp.data import Instance
+from allennlp.predictors import Predictor
+
+from allennlp_rc.eval import SquadEmAndF1
+from tqdm import tqdm
+
+logger = logging.getLogger(__name__)
+
+if __name__ == "__main__":
+    import allennlp_rc  # noqa F401: Needed to register the registrables.
+    import argparse
+
+    logging.basicConfig(level=logging.INFO)
+
+    parser = argparse.ArgumentParser(description="Evaluation for SQuAD 1.1")
+    parser.add_argument("--cuda-device", type=int, default=-1)
+    parser.add_argument("--qa-model", type=str)
+    parser.add_argument("--input-file", type=str)
+    args = parser.parse_args()
+
+    # Read inputs
+    check_for_gpu(args.cuda_device)
+    predictor = Predictor.from_path(
+        args.qa_model, predictor_name="transformer_qa", cuda_device=args.cuda_device
+    )
+    instances = predictor._dataset_reader.read(args.input_file)
+    logger.info("Running on %d instances", len(instances))
+
+    # We have to make sure we put instances with the same qid all into the same batch.
+    def batch_instances_by_qid(instances: Iterable[Instance]) -> Iterable[List[Instance]]:
+        current_qid = None
+        current_batch = []
+        for instance in instances:
+            instance_qid = instance["metadata"]["id"]
+            if current_qid is None:
+                current_qid = instance_qid
+            if instance_qid == current_qid:
+                current_batch.append(instance)
+            else:
+                yield current_batch
+                current_batch = [instance]
+                current_qid = instance_qid
+        if len(current_batch) > 0:
+            yield current_batch
+
+    def make_batches(
```

Comment:
I don't think it's important for this code itself, but it might be useful to the library broadly if you could see if these fit into the iterator abstraction. Definitely "extra credit". I wouldn't worry about it too much if you're slammed with other stuff.

Reply:
With the iterators in flux, I don't want to embark on a project extending their use right now.